### ISSUE TITLE
[ENH] 

### ISSUE THREAD

**Issue by pvdbk** [elsa-workflows/elsa-core#5889]
### Enhancement Overview

The "Result" output of the "HTTP Endpoint" activity can not be process in a variable for the moment, because the underlying object (a `DefaultHttpResponse`) can not be serialized when the activity execution entity is built.

### Proposed Enhancement

A `JsonConverter` could be added to handle the `HttpResponse` objects. The developers could also have a clean way to add some custom `JsonConverter`, for another exotic types, without reimplementing `IPayloadSerializer`.

**Comment by sNakiex** [comment]
Hi, you should not need the direct Result output of a HTTP Endpoint. 
You have the following output parameters you can use to access the data from the HTTP Endpoint:
![image](https://github.com/user-attachments/assets/0395b7d1-b97b-4655-b863-4d1115d568ed)

**Comment by pvdbk** [comment]
> Hi, you should not need the direct Result output of a HTTP Endpoint, what is the reason you want to use the full result from the HTTP Endpoint? You have the following output parameters you can use to access the data from the HTTP Endpoint.

Right, but the "Result" output still exists (particularly for ours users), and there is still a lake of JsonConverter.

**Comment by jeanbaptistedalle** [comment]
In fact, some of our users tried to call an endpoint but they didn't know exactly what kind of result they will get so they used the "Result" output to get the whole object instead of one variable for each other outputs. 

But I agree, the others output are enough but if the "Result" output doesn't work, maybe it's better to remove it ?

**Comment by sfmskywalker** [comment]
Perhaps the real issue is that when a user tries to bind a **persistent** variable to the HttpRequest object, the system crashes catastrophically. So perhaps we need to fix two things:

1. Make the system more resilient so that at least it doesn't crash the application.
2. Implement validation: when the user tries to bind an output to a variable, where the output type is known to be non-serializable, the UI should prevent the user from doing so.

Although I suspect that only 0.1% of users might actually use the HttpRequest value, it might be good to keep it for those rare occasions when you need to access something from the HTTP request. Or perhaps we should consider removing it (or, to @sNakiex 's point that it cannot be removed due to the fact that it's an inherited property, designate it to some other value).
